### PR TITLE
shelter show page showing reviews, review form complete

### PR DESCRIPTION
**Pull request by madhalle**
Review controller created
Review form html created
Review display added to shelter show page

### CODE REVIEW

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -45,6 +45,14 @@ h1 {
   padding: 40px 15px 15px 10px;
 }
 
+h2 {
```
Comment (madhalle) [inline app/assets/stylesheets/application.css:48]: Reviews is h2

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+class ReviewsController < ApplicationController
+
+  def new
+    # @shelter = Shelter.find(params[:id])
```
Comment (madhalle) [inline app/controllers/reviews_controller.rb:4]: will clean up in next branch

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    @shelter_id = params[:shelter_id]
```
Comment (madhalle) [inline app/controllers/reviews_controller.rb:5]: this :shelter_id needs to match with the route. Where it used to be shelters/:id it is shelters/:shelter_id whenever it relates to reviews

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  end
+
+  def create
+    shelter = Shelter.find(params[:shelter_id])
+    review = shelter.reviews.create!(review_params)
+    redirect_to "/shelters/#{shelter.id}"
+  end
+  # def create
```
Comment (madhalle) [inline app/controllers/reviews_controller.rb:14]: will clean up in next branch

File: app/controllers/reviews_controller.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  #   else
+  #     flash[:notice] = "Failed to enter review information: Please enter all info"
+  #     redirect_to "/shelters/#{@shelter_id}/new_review"
+  #   end
+  # end
+
+  private
+  def review_params
+    params.permit(:title, :rating, :content, :image)
+    # params.require(:review).permit(:title, :rating, :content, :image)
```
Comment (madhalle) [inline app/controllers/reviews_controller.rb:32]: will clean up in next branch

File: app/controllers/shelters_controller.rb [hunk 6]
```diff
@@ -5,6 +5,7 @@ def index
 
   def show
     @shelter = Shelter.find(params[:id])
+    @reviews = Review.all
```
Comment (madhalle) [inline app/controllers/shelters_controller.rb:8]: shelter show page needs access to reviews

File: app/models/review.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+class Review < ApplicationRecord
+  validates_presence_of :title
+  validates_presence_of :rating
+  validates_presence_of :content
+
+  belongs_to :shelter
```
Comment (madhalle) [inline app/models/review.rb:6]: relationship

Comment (madhalle) [reply app/models/review.rb:6]: one to many

File: app/models/shelter.rb [hunk 8]
```diff
@@ -1,5 +1,6 @@
 class Shelter < ApplicationRecord
   has_many :pets, dependent: :destroy
+  has_many :reviews
```
Comment (madhalle) [inline app/models/shelter.rb:3]: one to many

File: config/routes.rb [hunk 9]
```diff
@@ -18,4 +17,10 @@
   patch '/pets/:id', to: 'pets#update'
   delete '/pets/:id', to: 'pets#destroy'
   get '/pets/:id', to: 'pets#show'
+
+  get '/shelters/:shelter_id/new_review', to: 'reviews#new'
```
Comment (madhalle) [inline config/routes.rb:21]: :shelter_id relates to reviews

File: db/migrate/20200512210422_create_reviews.rb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+class CreateReviews < ActiveRecord::Migration[5.1]
+  def change
+    create_table :reviews do |t|
+      t.string :title
+      t.integer :rating
+      t.string :content
+      t.string :image
+      t.references :shelter, foreign_key: true
```
Comment (madhalle) [inline db/migrate/20200512210422_create_reviews.rb:8]: review model with foreign key to shelter

**Comment by melatran** [comment]
This looks great!

**Pull request merged**